src/ast_editor_state.mli

```diff
@@ -0,0 +1,34 @@
+open Import
+
+(** This module's interface is private to [Ast_editor.t]. It is in a separate
+    module to avoid circular dependencies with [Extension_instance.t] *)
+
+type t
+
+val make : unit -> t
+
+val find_original_doc_by_pp_uri : t -> Uri.t -> string option
+
+val find_webview_by_doc : t -> Vscode.Uri.t -> WebView.t option
+
+val set_changes_tracking : t -> TextDocument.t -> TextDocument.t -> unit
```

Comment:
I think we should change the API to take `Uri.t` everywhere. We only need `Uri.t` in all these functions so let's ask for only what we need.